### PR TITLE
Fixes #23926: Display group compliance

### PR DESCRIPTION
**Pull request by clarktsiory**
https://issues.rudder.io/issues/23926

Creates an Elm app for display of compliance by group (I copied most of the app structure from directives).
It also changes a bit of Scala frontend code in the summary page of a group, which therefore uses javascript to display compliance (this page has not been rewritten in Elm yet).

There is also some change in the Scala backend to add the policy mode of a rule and a node to the API response.

The 'global' and 'targeted' compliance is specific to groups, and it can be seen as filters for the rules and nodes displayed in the compliance tree : 
image
When the filter value is changed (e.g. from 'global' to 'targeted'), I made the compliance table display in "reload" state until the new compliance data is fetched, as it may take some time for the new data to be shown.

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/elm/sources/GroupCompliance/ViewNodesCompliance.elm [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  in
+    ( if model.ui.loading then
+      generateLoadingTable
+      else
+      div[]
+      [ div [class "table-header extra-filters"]
+        [ div [class "d-inline-flex align-items-baseline pb-3 w-25"]
+          [
+            div [class "btn-group yesno"]
+            [ label [class ("btn btn-default" ++ if isGlobalMode then " active" else ""), style "box-shadow" (if isGlobalMode then "inset 0 3px 5px rgba(0,0,0,.125)" else "none"), onClick (LoadCompliance GlobalCompliance)]
```
Comment (clarktsiory) [inline webapp/sources/rudder/rudder-web/src/main/elm/sources/GroupCompliance/ViewNodesCompliance.elm:53]: The styling is done "by-hand" here for testing, it is currently broken since the [bootstrap 5 migration](https://github.com/Normation/rudder/pull/5169), the shadow is no longer present when the button is active...

(This should be fixed globally in another PR)

**Comment by clarktsiory** [comment]
PR updated with a new commit

**Comment by clarktsiory** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]
